packages/patternfly-4/react-core/src/components/Select/Select.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import styles from '@patternfly/patternfly-next/components/Select/select.css';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+import SingleSelect from './SingleSelect';
+import SelectToggle from './SelectToggle';
+import { SelectContext } from './selectConstants';
+
+// seed for the aria-labelledby ID
+let currentId = 0;
+
+const propTypes = {
+  /** Content rendered inside the Select */
+  children: PropTypes.node,
+  /** Classes applied to the root of the Select */
+  className: PropTypes.string,
+  /** Flag to indicate if select is expanded */
+  isExpanded: PropTypes.bool,
+  /** Placeholder text of Select */
+  placeholderText: PropTypes.string,
+  /** Array of SelectOption nodes that will be rendered */
+  selectOptions: PropTypes.array,
+  /** Selected item(s) structure */
+  selections: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
+  /** Callback for selection behavior */
+  onSelect: PropTypes.func.isRequired,
+  /** Callback for toggle button behavior */
+  onToggle: PropTypes.func.isRequired,
+  /** Variant of rendered Select */
+  variant: PropTypes.oneOf(['single']),
+  /** Additional props are spread to the container <ul> */
+  '': PropTypes.any
+};
+
+const defaultProps = {
+  children: null,
+  className: '',
+  isExpanded: false,
+  selectOptions: null,
+  selections: null,
+  placeholderText: 'Please choose one',
+  variant: 'single'
+};
+
+class Select extends React.Component {
+  parentRef = React.createRef();
+  state = { openedOnEnter: false };
+
+  onEnter = () => {
+    this.setState({ openedOnEnter: true });
+  };
+
+  onClose = () => {
+    this.setState({ openedOnEnter: false });
+  };
+
+  render() {
+    const {
+      children,
+      className,
+      variant,
+      onToggle,
+      onSelect,
+      isExpanded,
+      selectOptions,
+      selections,
+      placeholderText,
+      ...props
+    } = this.props;
+    const { openedOnEnter } = this.state;
+    return (
+      <div className={css(styles.select, isExpanded && styles.modifiers.expanded, className)} ref={this.parentRef}>
+        <SelectContext.Provider value={onSelect}>
+          {variant === 'single' && (
+            <React.Fragment>
+              <SelectToggle
+                id={`pf-toggle-id-${currentId++}`}
```

Comment:
this id should be passed to the SingleSelect Component for the aria-labeledby prop.

Reply:
Is there an aria-labeledby on both SingleSelect and SelectToggle? Your message on SelectToggle makes is sound like the aria-labeledby is only on SelectToggle (so the id should be generated there).

Reply:
Looks like it is on both in core.  I am not sure what the SelectToggle aria-labeleld by is supposed to be linked to.  It is unclear to me in the core example.  @jgiardino do you know?

Reply:
Assuming that a label is present—like "Title" for the example included in the preview—the aria-labelledby attributes would be defined like this for the following elements:

- toggle: aria-labelledby="[id of Title label][id of toggle]"
- listbox: aria-labelledby="[id of Title label]"

Reply:
There is a hidden span in core that the aria-labelledby of the toggle and listbox reference, but I'm not sure what actually would get filled there. Should this hidden span label change to the currently selected value?

Reply:
The hidden span is intended to provide a label for the menu, similar to how a native `<select>` would have a label in a form. Since the toggle gets updated with the selected value, the `aria-labelledby` attribute references both the hidden label and the selected value to provide similar information that you'd hear with a native `<select>`. 

That said, I'm not sure what the best approach for implementing this should be. Consumers might want variations where a label is visible and variations where the label is not. And now I'm wondering if that label would be included separate from the component, so that the consumer has more control over how the label is presented, if visible. If that's the case, the consumer would need to specify `aria-labelledby` to reference the id of the toggle and the id of the label, visible or not. Is it possible for the consumer to just pass in the id of the label, and have the component automatically add `aria-labelledby="[id passed by consumer] [id generated for toggle by the component]"?